LINUX INITRD
############

Patch to support initrd loaded by kernelloader.

Kernel configuration for initrd support:
Deactivate "Machine selection"  ---> "Support for old style boot information"
Activate "Block devices" ---> "RAM disk support" (choose "*" and not "M")
Activate "Block devices" ---> "Initial RAM disk (initrd) support"

You should also activate a file system:
Activate "File systems" ---> "Second extended fs support" (choose "*" and not "M")

diff -ur --exclude .depend linux-2.4.17_ps2.orig/arch/mips/Makefile linux-2.4.17_ps2/arch/mips/Makefile
--- linux-2.4.17_ps2.orig/arch/mips/Makefile	2002-11-20 13:14:52.000000000 +0100
+++ linux-2.4.17_ps2/arch/mips/Makefile	2006-08-16 00:26:22.000000000 +0200
@@ -113,7 +113,7 @@
 # You need a compressed ramdisk image, named ramdisk.gz in
 # arch/mips/ramdisk
 #
-ifdef CONFIG_BLK_DEV_INITRD 
+ifdef CONFIG_EMBEDDED_RAMDISK
 CORE_FILES      += arch/mips/ramdisk/ramdisk.o
 SUBDIRS		+= arch/mips/ramdisk
 endif
diff -ur --exclude .depend linux-2.4.17_ps2.orig/arch/mips/ps2/prom.c linux-2.4.17_ps2/arch/mips/ps2/prom.c
--- linux-2.4.17_ps2.orig/arch/mips/ps2/prom.c	2002-07-10 08:14:52.000000000 +0200
+++ linux-2.4.17_ps2/arch/mips/ps2/prom.c	2006-08-16 00:30:48.000000000 +0200
@@ -40,6 +40,10 @@
 extern int akmem_bootinfo_size;
 #endif /* CONFIG_AKMEM */
 
+#ifdef CONFIG_BLK_DEV_INITRD
+extern unsigned long initrd_start, initrd_end;
+#endif
+
 void __init prom_init(int argc, char **argv, char **envp)
 {
 	struct ps2_bootinfo *bootinfo;
@@ -106,6 +110,11 @@
 	       bootinfo, oldbootinfo ? "(old style)" : "");
 	printk("boot option string at %p: %s\n",
 	       ps2_bootinfo->opt_string, arcs_cmdline);
+
+#ifdef CONFIG_BLK_DEV_INITRD
+	initrd_start = ps2_bootinfo->initrd_start;
+	initrd_end = initrd_start + ps2_bootinfo->initrd_size;
+#endif
 }
 
 void __init prom_free_prom_memory(void)
diff -ur --exclude .depend linux-2.4.17_ps2.orig/include/asm-mips/ps2/bootinfo.h linux-2.4.17_ps2/include/asm-mips/ps2/bootinfo.h
--- linux-2.4.17_ps2.orig/include/asm-mips/ps2/bootinfo.h	2002-04-09 08:08:04.000000000 +0200
+++ linux-2.4.17_ps2/include/asm-mips/ps2/bootinfo.h	2006-08-16 00:29:01.000000000 +0200
@@ -56,6 +56,8 @@
     char		*ver_dvd_rom;
     char		*ver_dvd_hdd;
     char		*ver_dvd_path;
+    uint32_t            initrd_start;
+    uint32_t            initrd_size;
 };
 #define PS2_BOOTINFO_OLDSIZE	((int)(&((struct ps2_bootinfo*)0)->magic))
 
